code/modules/projectiles/guns/projectile/sniper.dm

```diff
@@ -65,3 +65,51 @@
 	if(!bolt_open)
 		return
 	..()
+
+/obj/item/weapon/weaponparts
+	name = "weaponpart"
+	desc = "how did you get it?"
+	icon = 'icons/obj/buildingobject.dmi'
+
+/obj/item/weapon/weaponparts/heavysniper/disassembled
+	name = "disassembled sniper rifle"
+	desc = "This is a disassembled sniper rifle. You need to attach the stock."
+	icon_state = "sniperd"
+
+/obj/item/weapon/weaponparts/heavysniper/stocked
+	name = "disassembled sniper rifle with stock"
+	desc = "This is a disassembled sniper rifle with stock. Now attach the barrel."
+	icon_state = "snipersd"
+
+/obj/item/weapon/weaponparts/heavysniper/stock
+	name = "sniper rifle stock"
+	desc = "This is a stock from a sniper rifle."
+	icon_state = "sniperstock"
+
+
+/*/obj/item/weapon/weaponparts/heavysniper/optical_sight
```

Comment:
Don't bring things like this to our codebase, we don't need your commented out code. Just remove it.